### ISSUE TITLE
[BUG] Validation with must_get_agent_activity results in "awaiting deps" when called by the target agent

### ISSUE THREAD

**Issue by mattyg** [holochain/holochain#2916]
**Describe the bug**
When I test a validation function that includes a `must_get_agent_activity` it fails with:

```
 CellError(WorkflowError(SourceChainError(InvalidCommit("Awaiting deps [AnyDhtHash(uhCEkG0fcHonc2S31FZBmO15C7B2dbgeqRiVUPz77GC3PTkIJdQ3J)] but this is not allowed when committing entries to the source chain"))))'
 ```
 
 My test has only one agent, and it is that agent whose activity I'm must_get-ing. The must get call should not need to await deps, since the agent already has their own source chain.
 

**Expected behavior**
A must_get_agent_activity call should successfully get all deps, when called by the agent whose activity we're getting.
 

**System information:**
 - OS: Ubuntu 22.04
 - Holochain and HDK Version 0.2.2
 
**Steps to reproduce**
https://github.com/mattyg/bug-reproduction-conditional-validation-sweettest

**Comment by ThetaSinner** [comment]
Oh interesting, I wonder if this explains some of the log messages I've been seeing. So we aren't actually struggling to get dependencies in some cases, Holochain is just confused.

@abe-njama This looks like one to prioritise to me

**Comment by jost-s** [comment]
I believe this is fixed in `develop-0.2` with #2906. That commit fixed that must_get_agent_activity doesn't look in the agent's source chain.

**Comment by mattyg** [comment]
Confirmed that this is resolved by #2906
